.github/workflows/perf.yml

```diff
@@ -0,0 +1,69 @@
+name: perf
+
+# build on c/cpp changes or workflow changes
+on:
+  push:
+    paths:
+      - 'lib/**.[ch]'
+      - 'lib/**.cpp'
+      - 'src/**.[ch]'
+      - 'src/**.cpp'
+      - '**/CMakeLists.txt'
+      - 'cmake/Modules/**'
+      - 'util/buildbot/**'
+      - 'util/ci/**'
+      - '.github/workflows/**.yml'
+      - 'Dockerfile'
+      - '.dockerignore'
+# IMPORTANT: Do not run this workflow on pull request since this workflow
+# has permission to modify contents.
+
+permissions:
+  # deployments permission to deploy GitHub pages website
+  deployments: write
+  # contents permission to update benchmark contents in gh-pages branch
+  contents: write
+
+jobs:
+  # Current gcc version
+  perf_gcc_10:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install deps
+        run: |
+          source ./util/ci/common.sh
+          install_linux_deps g++-10
+
+      - name: Build
+        run: |
+          ./util/ci/build.sh
+        env:
+          CC: gcc-10
+          CXX: g++-10
+          CMAKE_FLAGS: "-DBUILD_BENCHMARKS=TRUE"
+
+      - name: Run Benchmarks
+        run: |
+          ./bin/minetest --run-benchmarks | tee output.txt
+
+      # gh-pages branch is updated and pushed automatically with extracted benchmark data
```

Comment:
Isn't this missing the part where it downloads previous benchmark results from cache?

Reply:
I followed this template:

https://github.com/marketplace/actions/continuous-benchmark#charts-on-github-pages-1

Past data is pulled from and re-written to the gh-pages .js file instead of a cache file.